src/Apps/cudaPlanesweepMAPTk.cpp

```diff
@@ -414,8 +414,9 @@ int main(int argc, char* argv[])
                 farZ = zs[(int)((zs.size()-1)*(1-outlierThres))];
                 std::cout<<"    distances from camera are "<<zs[0]<<"->"<<zs[zs.size()-1]<<" ("<< zs.size()<<" visible points)"<<std::endl;
                 std::cout<<"    distances ["<<outlierThres*100<<"%->"<<(1.0-outlierThres)*100<<"%] are "<<nearZ<<"->"<<farZ<<std::endl;
-                nearZ *= (1-safeness_margin_factor);
-                farZ *= (1+safeness_margin_factor);
+                const double safeness_margin = safeness_margin_factor * (farZ - nearZ);
+                nearZ -= safeness_margin;
+                farZ += safeness_margin;
```

Comment:
Agreed on the capping the nearZ at 0.

Using a percentage of the robust span is better than a percentage of the robust distance for long range scenes.  In some of my data the span of depths is something like 67.8 - 69.2.  So the range is only about 1.4, but the distance is more than an order of magnitude larger.  The current code expands my range to 44.9 - 92.2 which make the depth span 43.7, over four times bigger than it was.

Expanding relative to the depth gives me and expanded range of 67.3 - 69.7, which is much more reasonable.